### ISSUE TITLE
Issue with multiple file validation

### ISSUE THREAD

**Issue by robscott95** [wtforms/flask-wtf#393]
Hey, I'm having trouble with validating a `wtforms.MultipleFileField` field. 

Creating a form field like so:
`not_important = wtforms.MultipleFileField("Multiple file test",[FileRequired("File required"), FileAllowed(['csv'], ".csv file required")])` doesn't really work. After passing one or more files to the respective field, the validation returns "File required" error. 
But the multi-dict of `request.files` shows that all files are read correctly.
Sample:
```ImmutableMultiDict([('not_important', <FileStorage: 'Headline_all_test.csv' ('application/vnd.ms-excel')>)]```

So basically the issue is - flask requests reads files passed, but the `form.validate_on_submit()` errors out telling me, that no file has been passed.

BTW - found this fresh SO thread pertaining to the same issue: https://stackoverflow.com/questions/59734277/filerequiredvalidator-doesnt-work-when-using-multiplefilefield-in-my-form

**Comment by anshumankmr** [comment]
Hi, I am the OP from the StackOverflow question. I had tried this approach which seems to work for now.
`"""
Validation Class for RoomUpload
"""
class MyRoomValidator(object):
    def __init__(self, message=None):
        if not message:
            message = u"You need to upload the images with the category as well"
        self.message = message

    def __call__(self, form, field):
        print(len(form.roomImage.data))
        print((form.roomTypeSelect.data))
        if (not form.roomImage.data and  form.roomTypeSelect.data == -1) or(form.roomImage.data and  form.roomTypeSelect.data == -1) or (not form.roomImage.data and  form.roomTypeSelect.data != -1):#check for all possible combinations
            raise ValidationError(self.message)

class RoomUpload(FlaskForm):
    """
    This is meant for the sub-form for room details upload
    containing the details for room and its subtype

    roomImage:Upload the room image
    roomTypeSelect:The subcategory for room category

    The Form will only be submitted if the images  are uploaded and
    the RoomType will be selected
    """
    def validate_roomTypeSelect(form, field):#Inline Validator to ensure if default choice is not chosen
       print(field.data)
       if field.data == -1:
            raise ValidationError('Need to Select Room Type')
    def validate_roomImage(form,field):
        for roomFile in field.data:
            print(roomFile.filename)
            if  isanImageFile(roomFile.filename) == False: 
                raise ValidationError("Error!!! Not an Image File ")
    roomImage = MultipleFileField(u"Room",validators=[MyRoomValidator(),FileAllowed],id = "roomImage")
    roomTypeSelect= SelectField("Room Type",choices=roomTypes,coerce=int,id="roomTypeSelect")
    roomSubmit = SubmitField("Upload the images  of the room")`

and associated with this function route:
`@app.route('/upload_room_image',methods=['GET','POST'])
def upload_room_image():
    roomupload = RoomUpload()
    if request.method == 'POST' and roomupload.validate():
        saveImageWithSubCategory(roomupload,roomupload.roomImage)
        return (redirect(url_for("upload_image")))#redirect to upload sucess page, request.url is temporary make a sucess.html
    else:
        return(roomupload.errors) `

**Comment by HenriqueLin** [comment]
I think there is already an issue #337 about this problem and also a PR #338 by @greyli to solve this, but some how hasn't been merged yet. 😢

**Comment by snunzi** [comment]
I'm having the same issue, any work around?

**Comment by HenriqueLin** [comment]
> I'm having the same issue, any work around?

I think you can create your own field and validation by copy the code from PR #338.

**Comment by greyli** [comment]
I will try to implement what [David recommended](https://github.com/wtforms/flask-wtf/pull/338#issuecomment-394225278) late this month as an alternative to #338.

**Comment by damianpawlikowski** [comment]
I've uploaded validators for the `wtforms.fields.MultipleFileField` on my Github: https://github.com/damianpawlikowski/flaskwtf-boost. I may add it as PyPI tomorrow if there will be an interest.

Validators:
`flaskwtf_boost.validators.FilesRequired`
Allows you to set minimum and maximum count of required files, and forcing required param on the field if necessary.
`flaskwtf_boost.validators.FilesAllowed`
Same as `flask_wtf.file.FileAllowed` - works for list of an extensions and UploadSet(Note: use `pip install Flask-Reuploaded` or there may be an issues).
`flaskwtf_boost.validators.FilesSize`
Allows you to set minimum and maximum size for an independent file and minimum, maximum total size for an all uploads.

I've used it in the project sometime ago. You may just copy `flaskwtf_boost.validators.py` file somewhere in your project, for examples to the `utils.py` and import them and it should work.

Damian

**Comment by iROCKBUNNY** [comment]
@greyli Glad to here any progress in the future. Hope you are doing well. Now I just copy your old code and create my own field and validation as suggested by @HenriqueLin .

**Comment by khanfarhan10** [comment]
We really need this feature in flask wtforms, Any updates as to potentially when this could be updated please?

@greyli  @HenriqueLin @robscott95

**Comment by azmeuk** [comment]
Fix released in v1.2.0
